### PR TITLE
allow to override prefers-color-scheme via url param for guests

### PR DESCRIPTION
**Pull request by jonathanmmm**
using ?force-color-scheme=light should make the theme light and ?force-color-scheme=dark should make it dark.
Useful for iframes to match a website theme and to send a specific theme to a guest.

I couldn't test it locally, as it seems that javascript is somehow like compiled (and the src folder is missing for example) and have no developer setup. If needed, I can set one up.

My first pull request to nextcloud, if you have feedback how to make it better next time, tell me.

### CODE REVIEW

**Comment by jonathanmmm** [comment]
Could solve this issue: https://github.com/nextcloud/server/issues/29699

**Comment by jonathanmmm** [comment]
Ok, somehow it doesn't do anything.
Seems to be that /server/apps/accessibility/lib/AppInfo/Application.php
`$userValues = ['dark'];

			$hash = md5(implode('-', $userValues));
			$linkToCSS = $urlGenerator->linkToRoute(self::APP_ID . '.accessibility.getCss', ['md5' => $hash]);
			\OCP\Util::addHeader('link', ['rel' => 'stylesheet', 'media' => '(prefers-color-scheme: dark)', 'href' => $linkToCSS]);
			\OCP\Util::addScript('accessibility', 'accessibilityoca');`

If I put into the third line `light` instead of `dark` it changes to light theme.
Testing in incognito mode.

Does somebody know, if `accessibilityoca.js` in src has to be compiled and lands then in the js folder?

Don't know how these js files with the super long lines work, are they compiled or do I have to change them too?

This pull and commit seem to made prefer-color-scheme possible in the first place https://github.com/nextcloud/server/pull/21366/files
https://github.com/nextcloud/server/commit/ede5db10c3d5636305d5e9f5a9b41df8741d3e85

File: apps/accessibility/src/accessibilityoca.js [hunk 1]
```diff
@@ -24,8 +24,19 @@
...
+var force_color_scheme = srv_url.searchParams.get("force-color-scheme")
+if(force_color_scheme !== null) {
+	if(force_color_scheme=="dark") {
+		OCA.Accessibility.theme = 'dark'
+	}
+	else {
+		OCA.Accessibility.theme = 'light'
+	}
+}
+else if (OCA.Accessibility.checkMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (CarlSchwan) [inline apps/accessibility/src/accessibilityoca.js:38]:
```suggestion
const srvUrl = new URL(window.location.href)
const forceColorScheme = srvUrl.searchParams.get("force-color-scheme")
if (forceColorScheme !== null && (forceColorScheme === 'dark' || forceColorScheme === 'light')) {
    OCA.Accessibility.theme = forceColorScheme;
} else if (OCA.Accessibility.checkMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

A bit nicer :)

Comment (jonathanmmm) [reply apps/accessibility/src/accessibilityoca.js:38]: You are right, could you push it to my branch (have given access to reviewers/maintainers)? Or should I do it?

**Comment by jonathanmmm** [comment]
@CarlSchwan If you have motivation+time, we could takle this feature together.

**Comment by jonathanmmm** [comment]
Research:

https://github.com/nextcloud/server/commit/291640273dfc17aa240a48d7d4a1553d76d54e41
override these changes in the commit if url parameter is given

and change it so that line 27-30 the value in line 29 is not changed if url parameter is given
https://github.com/nextcloud/server/blob/master/apps/accessibility/src/accessibilityoca.js

**Comment by jonathanmmm** [comment]
@CarlSchwan It is working now when dark mode is enabled in the system (forcing light mode), but not the other way around and I don't know. I even tried and replaced every `theme--light` string I could find under accessibility and replaced it with `theme--dark`, still light mode.

**Comment by skjnldsv** [comment]
As there is no feedback since a while I will close this ticket. 
If you will decide to work on this feature again and if it hasn't been fixed or implemented already, feel free to re-open and solve the various conflicts.

Thanks for the interest in Nextcloud and the effort put into this! 🙇

**Pull request closed without merging**